### PR TITLE
GradesView

### PR DESCRIPTION
**Pull request by eduardcaragea27**
### Summary
UIGrades
### Metadata
- Ticket link: [UI Grades screen](https://trello.com/c/UmQWGbeL/6-ui-grades-screen-xib)
- Relevant design:

### Checks
- [ ] Team members are added to review the Pull Request
- [ ] Dead/unused code is deleted?
- [ ] PR does not break existing functionality
- [ ] Implementation matches the description in the story
- [ ] Images/Gifs have been added for all UI related changes

### CODE REVIEW

File: Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/CustomTableViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    init(course: Course) {
+        iconImageView.image = UIImage(named: course.iconName)
+        title.text = course.title
+        title.font = title.font.withSize(20)
+        grade.text = course.grade
+        grade.font = UIFont.systemFont(ofSize: 20, weight: .bold)
+        containerView.layer.cornerRadius = 20
+        containerView.layer.backgroundColor = .init(red: 0.751, green: 0.936, blue: 0.954, alpha: 1)
+        
+
```
Comment (iOSinternshipen) [inline Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/CustomTableViewCell.swift]: remove extra empty lines

File: Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/CustomTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+extension CustomTableViewCell {
+    func applyLayout() {
+        
+        contentView.layoutMargins = UIEdgeInsets(top: 8, left: 16, bottom: 8, right: 16)
+        
+        containerView.snp.makeConstraints { make in
+            make.top.trailing.leading.bottom.equalTo(contentView.layoutMargins)
+        }
+
```
Comment (iOSinternshipen) [inline Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/CustomTableViewCell.swift:48]: extra empty line

File: Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/GradesViewController.swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  
+    @IBOutlet weak var table: UITableView!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        table.dataSource = self
+        table.delegate = self
+        table.separatorStyle = .none
+       
+
```
Comment (iOSinternshipen) [inline Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/GradesViewController.swift]: extra empty lines

File: Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/CustomTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        grade.snp.makeConstraints { make in
+            make.trailing.equalToSuperview().inset(16)
+            make.centerY.equalTo(title.snp.centerY)
+            
+        }
+        
+        
+        
+        
+
```
Comment (iOSinternshipen) [inline Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/CustomTableViewCell.swift]: extra empty lines

File: Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/GradesViewController.swift [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        return data.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let inf = data[indexPath.row]
+        let courses = Course(iconName: inf.iconName, title: inf.title, grade: inf.grade)
+        
+        return CustomTableViewCell(course: courses)
+
+
```
Comment (iOSinternshipen) [inline Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/GradesViewController.swift]: extra empty lines

File: Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/GradesViewController.swift [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let inf = data[indexPath.row]
+        let courses = Course(iconName: inf.iconName, title: inf.title, grade: inf.grade)
+        
+        return CustomTableViewCell(course: courses)
+
+        
+        }
+
```
Comment (iOSinternshipen) [inline Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/GradesViewController.swift]: extra empty line

**Comment by carmenj** [comment]
Please update the PR template with the required info

File: Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/CustomTableViewCell.swift [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    let grade = UILabel()
+    
+    init(course: Course) {
+        iconImageView.image = UIImage(named: course.iconName)
+        title.text = course.title
+        title.font = title.font.withSize(20)
+        grade.text = course.grade
+        grade.font = UIFont.systemFont(ofSize: 20, weight: .bold)
+        containerView.layer.cornerRadius = 20
+        containerView.layer.backgroundColor = .init(red: 0.751, green: 0.936, blue: 0.954, alpha: 1)
```
Comment (iOSinternshipen) [inline Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/CustomTableViewCell.swift]: Implement UIColor extension

File: Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/CustomTableViewCell.swift [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+        applyLayout()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+extension CustomTableViewCell {
+    func applyLayout() {
```
Comment (IgnatCamelia) [inline Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/CustomTableViewCell.swift:42]: Can be private

Comment (MunBeniamin) [reply Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/CustomTableViewCell.swift:42]: Or the extensions to be private

File: Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/GradesViewController.swift [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        table.delegate = self
+        table.separatorStyle = .none
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return data.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let inf = data[indexPath.row]
```
Comment (MunBeniamin) [inline Internship2022TeamPatri/Internship2022TeamPatri/Controllers/Home/Grades/GradesViewController.swift]: What does "inf" mean? please don't use abbreviations.

**Review by MunBeniamin — APPROVED** [review]
Looks good now.

**Review by carmenj — APPROVED** [review]

**Pull request merged**